tests/MassTransit.Containers.Tests/Stop_Specs.cs

```diff
@@ -58,4 +58,66 @@ public Task Consume(ConsumeContext<PingMessage> context)
             }
         }
     }
+
+
+    [TestFixture]
+    public class Stop_Redelivery_Specs :
+        InMemoryTestFixture
+    {
+        [Test]
+        public async Task Should_respect_ConsumerStopTimeout()
+        {
+            var collection = new ServiceCollection();
+            collection.AddOptions<MassTransitHostOptions>().Configure(options => options.ConsumerStopTimeout = TimeSpan.FromSeconds(1));
+            collection.AddMassTransitTestHarness(configurator =>
+            {
+                configurator.AddTaskCompletionSource<ConsumeContext<PingMessage>>();
+                configurator.AddConsumer<StuckConsumer, StuckConsumerDefinition>();
+            });
+
+            var provider = collection.BuildServiceProvider(true);
+
+            var harness = provider.GetTestHarness();
+            await harness.Start();
+
+            await harness.Bus.Publish(new PingMessage(), harness.CancellationToken);
+
+            await provider.GetTask<ConsumeContext<PingMessage>>();
+
+            await harness.Stop(harness.CancellationToken);
+
+            using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
+            Assert.That(await harness.Consumed.Any<PingMessage>(x => x.Exception is OperationCanceledException, cts.Token), Is.True);
```

Comment:
Hmm, interesting. This should actually be `ArgumentException` which was originally thrown by the consumer.

Reply:
but do we "actually" notify this error to consumer? I guess it is waiting for retry cycle to complete, which is Cancelled

Reply:
I think you're right, it should remain as canceled to avoid moving any message to the error queue.